testindex/DependencyValidation.py

```diff
@@ -0,0 +1,126 @@
+import networkx as nx
+
+
+class ContainerDependencyGraph(object):
+    """This class encapsulates a networkx Directed Graph which is used to maintain
+    a dependency graph of containers"""
+
+    def __init__(self):
+        """Initializes an object of Container Dependency Graph"""
+        # Networkx represents each node as a list whose 1st element is a number to uniquely identify the node, and
+        # also help connect the edges, and the 2nd element is a dict which holds attributes which can be attached
+        # to each node.
+        self._dependency_graph = nx.DiGraph()
+        self._node_number = 1
+
+    def get_container_node(self, container_id=None, container_name=None):
+        """Gets the node, identified by container_name from the graph, if it exists."""
+        node = None
+        # Get the nodes from the networkx graph to iterate through
+        nodes = self._dependency_graph.nodes(data=True)
+        # Check if we need to get node by id or by name
+        # If neither is provided, raise exception
+        if not container_id and not container_name:
+            raise Exception("Either the id or the name of the container must be specified")
+        elif not container_id and container_name:
+            for item in nodes:
+                # Check the name label associated with the node to check if its the container_name we are looking for
+                if "name" in item[1] and container_name in item[1]["name"]:
+                    node = item
+                    break
+        # else if container id is provided, we just ignore container name
+        else:
+            for item in nodes:
+                # Check for matching id
+                if container_id == item[0]:
+                    node = item
+                    break
+        return node
+
+    def add_container(self, container_name, from_index=False, from_targetfile_baseimage=False):
+        """Checks if a node exists for container_name, and adds it if it does not exist"""
+        node = self.get_container_node(container_name=container_name)
+        if not node:
+            self._dependency_graph.add_node(self._node_number, name=container_name, from_index=from_index,
```

Comment:
line is seeming to have more than 80 characters, please take of pep8 guidelines.